### ISSUE TITLE
lammps will not install with intel compiler 17.0.4

### ISSUE THREAD

**Issue by ssergien** [LLNL/spack#4881]
lammps will not install with intel compiler 17.0.4. When I check spack-build.out, it is empty.

The only errors I see are:
```
==> Error: AttributeError: 'Spec' object has no attribute 'mpicxx_shared_libs'
/blues/gpfs/home/software/spack-0.10.1/var/spack/repos/builtin/packages/lammps/package.py:98, in edit:
     62       def edit(self, spec, prefix):
     63           config = []
     64   
     65           config.append('CC = c++')
     66           if self.compiler.name == 'intel':
     67               # This is taken from MAKE/OPTIONS/Makefile.intel_cpu_intelmpi
     68               config.append('OPTFLAGS = -xHost -O2 -fp-model fast=2 -no-prec-div -qoverride-limits')  # noqa: E501
     69               config.append('CCFLAGS = -g -qopenmp -DLAMMPS_MEMALIGN=64 -no-offload -fno-alias -ansi-alias -restrict $(OPTFLAGS)')  # noqa: E501
     70               config.append('LINKFLAGS = -qopenmp $(OPTFLAGS)')
     71           else:
     72               # This is taken from MAKE/OPTIONS/Makefile.g++
     73               config.append('OPTFLAGS = -O3')
     74               config.append('CCFLAGS = -fopenmp')
     75               config.append('LINKFLAGS = -fopenmp $(OPTFLAGS)')
     76   
     77           config.append('SHFLAGS = -fPIC')
     78           config.append('DEPFLAGS = -M')
     79           config.append('LINK = c++')
     80   
     81           config.append('LIB = ')
     82           config.append('SIZE = size')
     83   
     84           config.append('ARCHIVE = ar')
     85           config.append('ARFLAGS = -rc')
     86           config.append('SHLIBFLAGS = -shared')
     87   
     88           config.append('LMP_INC = -DLAMMPS_GZIP')
     89   
     90           mpi_path = self.spec['mpi'].prefix.lib
     91           mpi_inc = self.spec['mpi'].prefix.include
     92   
     93           config.append(
     94               'MPI_INC = -DMPICH_SKIP_MPICXX -DOMPI_SKIP_MPICXX=1 -I{0}'.format(
     95                   mpi_inc))
     96           config.append('MPI_PATH = -L{0}'.format(mpi_path))
     97           config.append('MPI_LIB = {0}'.format(
  >> 98               ' '.join(self.spec['mpi'].mpicxx_shared_libs)))
     99   
     100          config.append('FFT_INC = -DFFT_FFTW3 -L{0}'.format(
     101              self.spec['fftw'].prefix.include))
     102          config.append('FFT_PATH = -L{0}'.format(self.spec['fftw'].prefix.lib))
     103          config.append('FFT_LIB = -lfftw3')
     104  
     105          config.append('JPG_INC = ')
     106          config.append('JPG_PATH = ')
     107          config.append('JPG_LIB = ')
     108  
     109          makefile_inc_template = \
     110              join_path(os.path.dirname(self.module.__file__),
     111                        'Makefile.inc')
     112          with open(makefile_inc_template, "r") as fhr:
     113              config.extend(fhr.read().split('\n'))
     114  
     115          with working_dir('src/MAKE/'):
     116              with open('Makefile.{0}'.format(self.target_name), 'w') as fh:
     117                  fh.write('\n'.join(config))

See build log for details:
  /tmp/ssergien/spack-stage/spack-stage-I_NfTZ/lammps-stable_17Nov2016/spack-build.out
```

**Comment by davydden** [comment]
this is the error: `Error: AttributeError: 'Spec' object has no attribute 'mpicxx_shared_libs'`. Probably something is missing in IntelMPI package.

**Comment by davydden** [comment]
have a look at `openmpi` and see how `mpicxx_shared_libs` is defined.

**Comment by ssergien** [comment]
So here is what I did to make it work:

Change
```
mpi_path = self.spec['mpi'].prefix.lib
```
to
```
mpi_path = self.spec['mpi'].prefix.compilers_and_libraries.linux.mpi.lib64
```
Change
```
mpi_inc = self.spec['mpi'].prefix.include
```
to
```
mpi_inc = self.spec['mpi'].prefix.compilers_and_libraries.linux.mpi.include64
```
and change
```
config.append('MPI_LIB = {0}'.format(' '.join(self.spec['mpi'].mpicxx_shared_libs)))
```
to 
```
config.append('MPI_LIB = -lpthread -lmpi')
```
but I guess now that will not work with any other version of mpi. But these were a problem. For Intel MPI they point to wrong locations.

**Comment by davydden** [comment]
> Change...

it looks like you use external mpi. Due to convoluted lib structure of Intel MPI for now it only works when using Spack to install it. IIRC extra symlinks are done to make it work. Also see 
https://github.com/LLNL/spack/pull/4300

**Comment by alalazo** [comment]
`lammps` has been reworked recently by @junghans in #5313 Can you check if this issue still remains?

**Comment by junghans** [comment]
Yes, this should be fixed in the latest lammps spackage (effba3c95eae08dd704e3239887003134723173b).
